pkg/sharing/secret_reconciler.go

```diff
@@ -116,6 +116,19 @@ const (
 
 func (r *SecretReconciler) predictWantToReconcile(secret corev1.Secret) bool {
 	_, found := secret.Annotations[imagePullSecretAnnKey]
+	if found {
```

Comment:
sorry, just noticed i didn't take this stuff back out again;    will remove if we're keepign the above changes, otherwise probably just start a fresh branch anyhow.